Userland/Utilities/netstat.cpp

```diff
@@ -0,0 +1,204 @@
+    int state_column = -1;
+
+    auto add_column = [&](auto title, auto alignment, auto width) {
+        columns.append({ title, alignment, width, {} });
+        return columns.size() - 1;
+    };
+
+    protocol_column = add_column("Proto", Alignment::Left, 5);
+    bytes_in_column = add_column("Bytes-In", Alignment::Right, 9);
+    bytes_out_column = add_column("Bytes-Out", Alignment::Right, 9);
+    local_address_column = add_column("Local Address", Alignment::Left, 20);
+    peer_address_column = add_column("Peer Address", Alignment::Left, 20);
+    state_column = add_column("State", Alignment::Left, 9);
+
+    auto print_column = [](auto& column, auto& string) {
+        if (!column.width) {
+            out("{}", string);
+            return;
+        }
+        if (column.alignment == Alignment::Right) {
+            out("{:>{1}}  "sv, string, column.width);
+        } else {
+            out("{:<{1}}  "sv, string, column.width);
+        }
+    };
+
+    if (!has_protocol_flag || flag_tcp || flag_udp) {
+        out("Active Internet connections ");
+
+        if (flag_all)
+            outln("(servers and established)");
+        else {
+            if (flag_list)
+                outln("(only servers)");
+            else
+                outln("(without servers)");
+        }
+
+        for (auto& column : columns)
+            print_column(column, column.title);
+        outln();
+    }
+
+    if (!has_protocol_flag || flag_tcp) {
+        auto file = Core::File::construct("/proc/net/tcp");
+        if (!file->open(Core::IODevice::ReadOnly)) {
+            warnln("Error: {}", file->error_string());
+            return 1;
+        }
+
+        auto file_contents = file->read_all();
+        auto json = JsonValue::from_string(file_contents);
+        VERIFY(json.has_value());
+
+        Vector<JsonValue> sorted_regions = json.value().as_array().values();
+        quick_sort(sorted_regions, [](auto& a, auto& b) {
+            return a.as_object().get("local_port").to_u32() < b.as_object().get("local_port").to_u32();
+        });
+
+        for (auto& value : sorted_regions) {
+            auto if_object = value.as_object();
+
+            auto bytes_in = if_object.get("bytes_in").to_string();
+            auto bytes_out = if_object.get("bytes_out").to_string();
+            auto peer_address = if_object.get("peer_address").to_string();
+            auto peer_port = if_object.get("peer_port").to_string();
+            auto formatted_peer_address = String::formatted("{}:{}", peer_address, peer_port);
+            auto local_address = if_object.get("local_address").to_string();
+            auto local_port = if_object.get("local_port").to_string();
+            auto formatted_local_address = String::formatted("{}:{}", local_address, local_port);
+            auto state = if_object.get("state").to_string();
+
+            if (!flag_all && (((state == "Listen") && !flag_list) || (!(state == "Listen") && flag_list)))
```

Comment:
```suggestion
            if (!flag_all && ((state == "Listen" && !flag_list) || (state != "Listen" && flag_list)))
```

Reply:
Thanks for the feedback! Made suggested changes 👍